osu.Server.Spectator.Tests/Multiplayer/MultiplayerCountdownTest.cs

```diff
@@ -248,24 +251,26 @@ public async Task AutoStartCountdownDoesNotStartWithZeroDuration()
                 var room = usage.Item;
                 Debug.Assert(room != null);
 
-                Assert.False(room.IsCountdownRunning);
+                Assert.Null(room.Countdown);
             }
         }
 
-        [Fact]
+        [Fact(Timeout = 1000)]
         public async Task AutoStartCountdownStartsWhenHostReadies()
         {
             await Hub.JoinRoom(ROOM_ID);
             await Hub.ChangeSettings(new MultiplayerRoomSettings { AutoStartDuration = TimeSpan.FromMinutes(1) });
 
             await Hub.ChangeState(MultiplayerUserState.Ready);
-            await waitForCountingDown();
 
-            await finishCountdown();
+            using (var usage = await Hub.GetRoom(ROOM_ID))
+                Assert.NotNull(usage.Item?.Countdown);
```

Comment:
The countdown object is set on the async thread. Removing the `waitForCountingDown()` makes this assert potentially intermittent and testable by adding a `Task.Delay()` at the start of the async operation.

Reply:
Thanks, I knew there was some reason it was done like this. I've done a bit of refactoring, see what you think (seems to work now with `Task.Delay`s scattered in the flow).

Reply:
I think it feels better with the rewrite, it follows how I originally envisioned this method a bit more.